### PR TITLE
added provides-1 test

### PR DESCRIPTION
**Pull request by jrohel**
Tests for dnf provides command.
Only values provided by packages are tested not filenames in packages.

### CODE REVIEW

File: dnf-docker-test/features/provides-1.feature [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+         |              | Conflicts | TestC         |
+         | TestE        | Version   | 2             |
+         |              | Provides  | TestEVal      |
+       When I enable repository "base"
+        And I successfully run "dnf -y install TestA TestC TestD"
+        And I disable repository "base"
+
+  Scenario: dnf provides TestB - installed package TestD-1 provides TestB
+       When I successfully run "dnf provides TestB"
+       Then the command stdout should match regexp "TestD-1-[^R]+Repo[ \t]+: @System"
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/provides-1.feature]: why not compare whole output?

File: dnf-docker-test/features/provides-1.feature [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+         | TestC        | Version   | 2             |
+         |              | Provides  | TestB         |
+         |              | Conflicts | TestD         |
+         | TestD        | Version   | 2             |
+         |              | Provides  | TestB         |
+         |              | Conflicts | TestC         |
+         | TestE        | Version   | 2             |
+         |              | Provides  | TestEVal      |
+       When I enable repository "base"
+        And I successfully run "dnf -y install TestA TestC TestD"
```
Comment (ignatenkobrain) [inline dnf-docker-test/features/provides-1.feature]: you should compare rpmdb as well to ensure that proper things were installed

File: dnf-docker-test/features/provides-1.feature (hunk 1 above)
Comment (jrohel) [inline dnf-docker-test/features/provides-1.feature]: 1. The order of printing of packages is not defined. When I run tests more times the order differs.
2. Want we strictly test number of spaces (etc.)?

File: dnf-docker-test/features/provides-1.feature (hunk 2 above)
Comment (jrohel) [inline dnf-docker-test/features/provides-1.feature]: Good idea. I added comparison of rpmdb.

**Comment by j-mracek** [comment]
@ignatenkobrain Please could you finish the review?

**Comment by j-mracek** [comment]
All test passed, code looks fine. Thanks.

**Pull request merged**